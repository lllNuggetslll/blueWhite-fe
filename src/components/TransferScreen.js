import React, { useState } from "react";
import { connect } from "react-redux";
import { object, func } from "prop-types";
import Button from "@material-ui/core/Button";
import Input from "@material-ui/core/Input";
import Radio from "@material-ui/core/Radio";
import RadioGroup from "@material-ui/core/RadioGroup";
import FormControlLabel from "@material-ui/core/FormControlLabel";

import styled from "styled-components";

import { updateUserData, fetchUserData } from "./../actions/userActions";

const InputContainer = styled.div`
  margin-top: 20px;
  display: flex;
  width: 290px;
`;

const TransferScreen = ({ userInfo, updateUserData }) => {
  const [mode, setMode] = useState("deposit");
  const [amount, setAmount] = useState("");

  const onEnterAmount = e => {
    const { value } = e.target;

    setAmount(value);
  };

  const onUpdateUser = async () => {
    const { balance } = userInfo;
    let newTotal;

    if (mode === "deposit") {
      newTotal = Number.parseInt(balance + amount);
    } else {
      newTotal = Number.parseInt(balance - amount);
    }

    if (newTotal < 0) newTotal = 0;
    try {
      await updateUserData({ balance: newTotal });
      setAmount("");
    } catch (err) {
      console.log(err);
    }
  };

  const onToggleMode = e => {
    setMode(e.target.value);
  };

  return (
    <div>
      <h1>Transfer Funds</h1>
      <RadioGroup
        aria-label="mode"
        name="gender1"
        value={mode}
        onChange={onToggleMode}
      >
        <FormControlLabel value="deposit" control={<Radio />} label="Deposit" />
        <FormControlLabel
          value="transfer"
          control={<Radio />}
          label="Transfer"
        />
      </RadioGroup>
      <InputContainer>
        <Input
          type="number"
          onChange={onEnterAmount}
          value={amount}
          placeholder="dollar amount"
        />
        <Button onClick={onUpdateUser}>
          {mode === "deposit" ? "Deposit" : "Transfer"}
        </Button>
      </InputContainer>
    </div>
  );
};

TransferScreen.propTypes = {
  userInfo: object.isRequired,
  updateUserData: func.isRequired
};

export default connect(
  null,
  { updateUserData, fetchUserData }
)(TransferScreen);
