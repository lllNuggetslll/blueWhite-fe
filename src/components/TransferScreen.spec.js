import React from "react";
import { TransferScreen } from "./TransferScreen";
import { shallow } from "enzyme";

describe("Screen Container", () => {
  const requiredProps = {
    userInfo: {},
    updateUserData: () => {}
  };
  const wrapper = shallow(<TransferScreen {...requiredProps} />);

  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
