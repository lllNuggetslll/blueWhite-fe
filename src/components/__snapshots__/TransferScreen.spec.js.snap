// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Container should render 1`] = `
<div>
  <h1>
    Transfer Funds
  </h1>
  <ForwardRef(RadioGroup)
    aria-label="mode"
    name="gender1"
    onChange={[Function]}
    value="deposit"
  >
    <WithStyles(ForwardRef(FormControlLabel))
      control={<ForwardRef(WithStyles) />}
      label="Deposit"
      value="deposit"
    />
    <WithStyles(ForwardRef(FormControlLabel))
      control={<ForwardRef(WithStyles) />}
      label="Transfer"
      value="transfer"
    />
  </ForwardRef(RadioGroup)>
  <styled.div>
    <WithStyles(ForwardRef(Input))
      onChange={[Function]}
      placeholder="dollar amount"
      type="number"
      value=""
    />
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      Deposit
    </WithStyles(ForwardRef(Button))>
  </styled.div>
</div>
`;
